plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/service/QueryGroupPersistenceService.java

```diff
@@ -192,6 +194,21 @@ private Map<ResourceType, Double> calculateTotalUsage(Map<String, QueryGroup> ex
         return map;
     }
 
+    /**
+     * Get the QueryGroups with the specified name from cluster state
+     * @param name - the QueryGroup name we are getting
+     * @param currentState - current cluster state
+     */
+    public static List<QueryGroup> getFromClusterStateMetadata(String name, ClusterState currentState) {
+        Map<String, QueryGroup> currentGroups = currentState.getMetadata().queryGroups();
+        if (name == null || name.isEmpty()) {
+            return new ArrayList<>(currentGroups.values());
```

Comment:
Can we use more generic `Collection` interface instead of `List` to avoid this unnecessary conversion?